File: junior/contest2/prog/enclosure.py
#!/usr/bin/python3
import sys
import math

if __name__ == "__main__":
	for n in range(0,5):
		line = input().strip()
		exp = line.replace(' ','')
		res = []
		leftpos = exp.find('(')
		rightpos = exp.find(')')
		if leftpos >= 0 and rightpos < 0: # has left missing right
			for i in range(leftpos, len(exp)):
				if exp[i] == '+' or exp[i] == '-' or exp[i] == '*' or exp[i] == '/':
					res.append(i+3)
		elif leftpos < 0 and rightpos >= 0: # has right missing left
			for i in range(0, rightpos):
				if exp[i] == '+' or exp[i] == '-' or exp[i] == '*' or exp[i] == '/':
					res.append(i)
		j = 0
		for num in res:
			print(num, end = ", " if j < len(res)-1 else '\n')
			j += 1

File: junior/contest2/prog/enclosure.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
	for(int n = 0; n < 5; ++n) {
		string exp, line;
		getline(cin, line);
		for(char ch : line) // remove spaces
			if(ch != ' ')
				exp += ch;
		vector<int> res;
		auto leftpos = exp.find('(');
		auto rightpos = exp.find(')');
		bool foundleft = (leftpos != string::npos);
		if(foundleft) { // missing right
			for(int i = leftpos; i < exp.length(); ++i) {
				char c = exp[i];
				switch(c) {
				case '+':
				case '-':
				case '*':
				case '/':
					res.push_back(i+3);
					break;
				}
			}
		} else { // missing left
			for(int i = 0; i < rightpos; ++i) {
				char c = exp[i];
				switch(c) {
				case '+':
				case '-':
				case '*':
				case '/':
					res.push_back(i);
					break;
				}
			}
		}
		for(int i = 0; i < res.size(); ++i) {
			cout << res[i];
			if(i < res.size()-1)
				cout << ", ";
		}
		cout << endl;
	}
	return 0;
}
